test/e2e/self_hosted.go

```diff
@@ -396,6 +396,10 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 		if selfHostedNamespace != nil {
 			// Dump all Cluster API related resources to artifacts before pivoting back.
 			dumpAllResources(ctx, selfHostedClusterProxy, input.ArtifactFolder, namespace, clusterResources.Cluster)
+
```

Comment:
Independently. Looks like sometimes we use a "machine" folder when we call CollectWorkloadClusterLogs.

As far as I can tell this should be wrong as CollectWorkloadClusterLogs internally adds machine / machine-pools.

Can you check? Somehow I didn't find an example in a testgrid folder

Reply:
AFAICT we were never collecting those logs as the logger wasn't present in the workloadClusterProxy used for the management cluster.

I've added improvements there to this PR aswell.